packages/@aws-cdk/aws-apigatewayv2-integrations/lib/alb-integration.ts

```diff
@@ -0,0 +1,40 @@
+import { HttpMethod, IVpcLink } from '@aws-cdk/aws-apigatewayv2';
+import * as elbv2 from '@aws-cdk/aws-elasticloadbalancingv2';
+import { HttpProxyPrivateIntegration } from './http-private';
+
+/**
+ * Properties to initialize `ALBPrivateIntegration`.
+ */
+export interface ALBPrivateIntegrationProps {
+  /**
+   * The listener to the application load balancer used for the integration
+   */
+  readonly listener: elbv2.ApplicationListener;
+
+  /**
+   * The vpc link to be used for the private integration
+   *
+   * @default - a new VpcLink is created
+   */
+  readonly vpcLink?: IVpcLink;
+
+  /**
+   * The HTTP method that must be used to invoke the underlying HTTP proxy.
+   * @default HttpMethod.ANY
+   */
+  readonly method?: HttpMethod;
+}
+
+/**
+ * The Application Load Balancer Private integration resource for HTTP API
+ */
+export class ALBPrivateIntegration extends HttpProxyPrivateIntegration {
```

Comment:
Naming: 
1. We use pascal case for acronyms, so this should be 'Alb' and not 'ALB'.

2. Terser - Just `AlbIntegration` should be sufficient. The "private" feels redundant. Incorporating the previous comment, this will be `HttpAlbIntegration`.